spikeextractors/extractors/nwbextractors/nwbextractors.py

```diff
@@ -256,7 +256,7 @@ def get_traces(self, channel_ids=None, start_frame=None, end_frame=None):
         with NWBHDF5IO(self._path, 'r') as io:
             nwbfile = io.read()
             es = nwbfile.acquisition[self._electrical_series_name]
-            table_ids = [list(es.electrodes.data[:]).index(id) for id in channel_ids]
+            table_ids = [list(nwbfile.electrodes.id[:]).index(id) for id in channel_ids]
```

Comment:
```suggestion
            table_ids = [es.electrodes[:].index.tolist().index(chan) for chan in channel_ids]
```

The above code will work, but assumes that the ElectricalSeries is referencing all of the electrodes in the electrodes able. That will work for NWB files created by SpikeExtractor, but won't work in the general case, which may include `ElectricalSeries` that correspond to specific subsets of electrodes. This suggestion bases the indexing on the ids of the electrodes that are in the subregion of the electrodes table that is referenced by the target `ElectricalSeries`, so it should be more robust

Reply:
I'm trying this out and the line @bendichter suggests seems to work, but causes massive computation delays -- like 60 seconds delay -- when number of channels is 192 and duration is 115 minutes.

Whereas I confirm that the solution in the PR works but has no such delay. But as Ben points out, maybe that isn't sufficient for all cases.

Reply:
@magland thanks for the report. I know what's going on and I'm pretty sure I can fix it.

Reply:
@magland I have edited the suggestion. Can you try that and confirm that it works and does not cause computational delays for you?

Reply:
@bendichter - in your edit, where is the `ephys_ts` variable coming from? I tried to use `nwbfile` in place but got an error: 'DynamicTable' object has no attribute 'table'

Reply:
Oh I used `es` and that seems to have worked. So I have:
```
            es_channel_ids = np.array(es.electrodes.table.id[:])[es.electrodes.data[:]].tolist()
            table_ids = [es_channel_ids.index(id) for id in channel_ids]
```
And I confirm that it works, and is efficient. Thx @bendichter!

Reply:
@luiztauffer can you incorporate Ben's suggestion?